cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -410,6 +418,11 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 	).Run()
 	time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
 
+	// run the shared informers
+	for _, informer := range informers {
+		go informer.Run(wait.NeverStop)
```

Comment:
@saad-ali the internalPodInformer field of the controllers is only set when calling the New*ControllerFromClient() functions.  All of the Run() functions have a check to see if the internalPodInformer != nil before trying to start it.

Said another way, tests use New*ControllerFromClient() so that the Run() will start the informer.  The regular New*Controller() function, don't set internalPodInformer, and the shared informer is started here.

Also, FYI I have PR #24841 open for converting some of the remaining controllers to using the shared pod informer.